target/scripts/startup/setup.d/getmail.sh

```diff
@@ -4,38 +4,59 @@ function _setup_getmail() {
   if [[ ${ENABLE_GETMAIL} -eq 1 ]]; then
     _log 'trace' 'Preparing Getmail configuration'
 
-    local GETMAILRC ID CONFIGS
+    # Verify correct value for GETMAIL_POLL. Valid are any numbers greater than 0.
+    if ! [[ ${GETMAIL_POLL} =~ ^[0-9]+$ && ${GETMAIL_POLL} -gt 0 ]]; then
+      _log 'warn' "Invalid value for GETMAIL_POLL: ${GETMAIL_POLL}"
+      _log 'warn' "Getmail will be disabled"
+      # return 1
+    fi
 
-    GETMAILRC='/etc/getmailrc.d'
-    CONFIGS=0
+    local GETMAIL_RC ID GETMAIL_DIR
 
-    mkdir -p "${GETMAILRC}"
+    local GETMAIL_CONFIG_DIR='/tmp/docker-mailserver/getmail'
+    local GETMAIL_RC_DIR='/etc/getmailrc.d'
+    local GETMAIL_RC_GENERAL_CF="${GETMAIL_CONFIG_DIR}/getmailrc_general.cf"
+    local GETMAIL_RC_GENERAL='/etc/getmailrc_general'
 
-    # Generate getmailrc configs, starting with the `/etc/getmailrc_general` base config,
-    # Add a unique `message_log` config, then append users own config to the end.
-    for FILE in /tmp/docker-mailserver/getmail-*.cf; do
-      if [[ -f ${FILE} ]]; then
-        CONFIGS=1
-        ID=$(cut -d '-' -f 3 <<< "${FILE}" | cut -d '.' -f 1)
-        local GETMAIL_CONFIG="${GETMAILRC}/getmailrc-${ID}"
+    # Create the directory /etc/getmailrc.d to place the user config in later.
+    mkdir -p "${GETMAIL_RC_DIR}"
 
-        cat /etc/getmailrc_general                                 >"${GETMAIL_CONFIG}"
-        echo -e "message_log = /var/log/mail/getmail-${ID}.log\n" >>"${GETMAIL_CONFIG}"
-        cat "${FILE}"                                             >>"${GETMAIL_CONFIG}"
-      fi
-    done
+    # Check if getmail config directory exists and at least one <ID>.cf file is present.
+    # getmailrc_general.cf is not mandatory and excluded.
+    if ! find "${GETMAIL_CONFIG_DIR}" -type f -name '*.cf' -not -name getmailrc_general.cf 2>/dev/null | grep -q .; then
+      _log 'warn' 'No getmail configration found'
+      _log 'warn' "Getmail will be disabled"
+    fi
```

Comment:
> It says it will disable, but there is no action that does so, logic continues to run...?

See my other comment, where I tried to explain the why.

> Regarding the comment and extension filtered, my earlier suggestions with the docs omitted any expectation of .cf

I would continued to use .cf extension, because we use .cf for all config files in the config dir and I didn't want getmail to be an exception.